/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

From 6c4d82990973b02a756ab3ba83ea345fed4bf87e Mon Sep 17 00:00:00 2001
From: Stuart Kreitman <stuart.kreitman@sun.com>
Date: Wed, 27 Sep 2006 19:02:02 -0700
Subject: [PATCH] Sun bug #5099413/Xorg bugzilla #1554: Gnopernicus full-screen mag not working

Sun bug #5099413: Gnopernicus full-screen magnification not working
<http://bugs.opensolaris.org/bugdatabase/view_bug.do?bug_id=5099413>

includes fix for Xorg bug #1554: rect defined in XFixesInvertRegion is not used
<https://bugs.freedesktop.org/show_bug.cgi?id=1554>
---
 src/Region.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/Region.c b/src/Region.c
index c9f5920..0bbc1a2 100644
--- src/Region.c
+++ src/Region.c
@@ -251,6 +251,10 @@ XFixesInvertRegion (Display *dpy, Xserve
     GetReq (XFixesInvertRegion, req);
     req->reqType = info->codes->major_opcode;
     req->xfixesReqType = X_XFixesInvertRegion;
+    req->x = rect->x;
+    req->y = rect->y;
+    req->width = rect->width;
+    req->height = rect->height;
     req->source = src;
     req->destination = dst;
     UnlockDisplay (dpy);
@@ -320,6 +324,7 @@ XFixesFetchRegionAndBounds (Display	    
     req->reqType = info->codes->major_opcode;
     req->xfixesReqType = X_XFixesFetchRegion;
     req->region = region;
+    *nrectanglesRet = 0;
     if (!_XReply (dpy, (xReply *) &rep, 0, xFalse))
     {
 	UnlockDisplay (dpy);
-- 
1.4.1

